lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/security/RestFileBasedSystemAccessControlRulesProvider.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.base.security;
+
+import com.google.inject.Inject;
+import io.airlift.http.client.HttpClient;
+
+public class RestFileBasedSystemAccessControlRulesProvider
```

Comment:
same about inheritance

Reply:
Here I would need to get the `HttpClient` from somewhere to use the `toProvider` approach as described above. Any idea where to get that from? Is it even possible?

Reply:
We can instead use a custom `Provider` or use create a method in the parent module with `@Produces` annotation